lua/rsync/sync.lua

```diff
@@ -28,16 +29,25 @@ local function safe_sync(command, on_start, on_exit)
         on_stderr = function(_, output, _)
             -- skip when function reports empty error
             if vim.inspect(output) ~= vim.inspect({ "" }) then
-                log.info(string.format("safe_sync command: '%s', on_stderr: '%s'", command, vim.inspect(output)))
+                if config.values.log then
+                    log.info(string.format("safe_sync command: '%s', on_stderr: '%s'", command, vim.inspect(output)))
+                end
+
+                config.values.on_error(output, command)
             end
         end,
 
         -- job done executing
         on_exit = function(_, code, _)
             on_exit(code)
             if code ~= 0 then
-                log.info(string.format("safe_sync command: '%s', on_exit with code = '%s'", command, code))
+                if config.values.log then
```

Comment:
Since I wrapped this here anyways, and added callbacks, maybe I could completly move the `log.info` to the user callback, what do you think?